Spanner/src/Operation.php

```diff
@@ -187,6 +188,11 @@ public function commitWithResponse(Session $session, array $mutations, array $op
      */
     public function rollback(Session $session, $transactionId, array $options = [])
     {
+        // TODO: Check with team on this and decide
+        if (is_array($transactionId) and isset($transactionId['begin'])) {
```

Comment:
What is the use-case for `$transactionId` being an array? Since the function just returns in this case, and the phpdoc says `$transactionId` is a `string`, are we sure we want to support it being an array?

Reply:
Use-case is, users can provide transaction options for the `begin` and I wanted to preserve it.

We can do this by either pass it in the transactionId as shown here or we can make a new attribute in the transaction class to preserve the `begin` options.

Reply:
It sounds like passing the `$transactionId` is a different option than passing in `$beginTransaction` (or something along those lines), and they should indeed be two options